tests/components/sentry/test_init.py

```diff
@@ -0,0 +1,330 @@
+            custom_components=["ironing_robot"],
+            event={},
+            hint={},
+        )
+
+    assert result
+    assert result["tags"]["integration"] == "ironing_robot"
+    assert result["tags"]["platform"] == "switch"
+    assert result["tags"]["custom_component"] == "yes"
+
+
+@pytest.mark.parametrize(
+    "logger,tags",
+    [
+        ("adguard", {"package": "adguard"}),
+        (
+            "homeassistant.components.hue.coordinator",
+            {"integration": "hue", "custom_component": "no"},
+        ),
+        (
+            "homeassistant.components.hue.light",
+            {"integration": "hue", "platform": "light", "custom_component": "no"},
+        ),
+        (
+            "homeassistant.components.ironing_robot.switch",
+            {
+                "integration": "ironing_robot",
+                "platform": "switch",
+                "custom_component": "yes",
+            },
+        ),
+        (
+            "homeassistant.components.ironing_robot",
+            {"integration": "ironing_robot", "custom_component": "yes"},
+        ),
+        ("homeassistant.helpers.network", {"helpers": "network"}),
+        ("tuyapi.test", {"package": "tuyapi"}),
+    ],
+)
+async def test_logger_event_extraction(hass: HomeAssistant, logger, tags):
+    """Test extraction of information from Sentry logger events."""
+
+    result = process_before_send(
+        hass,
+        options={
+            CONF_EVENT_CUSTOM_COMPONENTS: True,
+            CONF_EVENT_THIRD_PARTY_PACKAGES: True,
+        },
+        channel="test",
+        huuid="12345",
+        system_info={"installation_type": "pytest"},
+        custom_components=["ironing_robot"],
+        event={"logger": logger},
+        hint={},
+    )
+
+    assert result
+    assert result["tags"] == {
+        "channel": "test",
+        "uuid": "12345",
+        "installation_type": "pytest",
+        **tags,
+    }
+
+
+@pytest.mark.parametrize(
+    "logger,options,event",
+    [
+        ("adguard", {CONF_EVENT_THIRD_PARTY_PACKAGES: True}, True),
+        ("adguard", {CONF_EVENT_THIRD_PARTY_PACKAGES: False}, False),
+        (
+            "homeassistant.components.ironing_robot.switch",
+            {CONF_EVENT_CUSTOM_COMPONENTS: True},
+            True,
+        ),
+        (
+            "homeassistant.components.ironing_robot.switch",
+            {CONF_EVENT_CUSTOM_COMPONENTS: False},
+            False,
+        ),
+    ],
+)
+async def test_filter_log_events(hass: HomeAssistant, logger, options, event):
+    """Test filtering of events based on configuration options."""
+    result = process_before_send(
+        hass,
+        options=options,
+        channel="test",
+        huuid="12345",
+        system_info={"installation_type": "pytest"},
+        custom_components=["ironing_robot"],
+        event={"logger": logger},
+        hint={},
+    )
+
+    if event:
+        assert result
+    else:
+        assert result is None
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)